engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java

```diff
@@ -1061,6 +1062,77 @@ public void shouldTerminateMultiInstanceBodyAndNestedElements() {
             tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATED));
   }
 
+  @Test
+  public void shouldTerminateFlowScopeWhenElementInsideIsActivateAndFlowScopeIsTerminated() {
+    // given
+    final Consumer<EventSubProcessBuilder> eventSubProcess =
+        esp ->
+            esp.startEvent("startEvent")
+                .message(m -> m.name("message").zeebeCorrelationKeyExpression("\"correlationKey\""))
```

Comment:
🔧 The engine is not cleaned up between the tests. To avoid that the message correlates to multiple process instances (from other tests), we can use `CLASS_RULE_HELPER.getCorrelationValue()` to get a unique correlation key per test method. Using different correlation keys between the tests makes sure the tests stay isolated.

Thanks @saig0 for pointing this one out to me today

Reply:
Good catch